packages/patrol/darwin/Classes/AutomatorServer/Automator/IOSAutomator.swift

```diff
@@ -885,6 +887,15 @@
       }
     }
 
+    func setMockLocation(latitude: Double, longitude: Double) throws {
+      if #available(iOS 16.4, *) {
+        DispatchQueue.main.async {
+          XCUIDevice.shared.location = XCUILocation(
+            location: CLLocation(latitude: latitude, longitude: longitude))
+        }
```

Comment:
I think you can use `runAction` method instead. It wraps the call with logging and calls it using `DispatchQueue.main.async`